### ISSUE TITLE
Cannot read property 'autoDetectRenderer' of undefined

### ISSUE THREAD

**Issue by dagatsoin** [Izzimach/react-pixi#69]
I am trying `react-pixi` but I have an error. Can someone help me on this ?
```
TypeError: Cannot read property 'autoDetectRenderer' of undefined
    at Object.generateDefaultRenderer (modules.js?hash=c8c86f2…:134428)
    at Object.componentDidMount (modules.js?hash=c8c86f2…:134440)
    at modules.js?hash=c8c86f2…:82477
    at measureLifeCyclePerf (modules.js?hash=c8c86f2…:82287)
    at modules.js?hash=c8c86f2…:82476
    at CallbackQueue.notifyAll (modules.js?hash=c8c86f2…:75298)
    at ReactReconcileTransaction.close (modules.js?hash=c8c86f2…:85548)
    at ReactReconcileTransaction.closeAll (modules.js?hash=c8c86f2…:76457)
    at ReactReconcileTransaction.perform (modules.js?hash=c8c86f2…:76404)
    at ReactUpdatesFlushTransaction.perform (modules.js?hash=c8c86f2…:76391)
```

Here are the version I use and the code of my component.

```
...
  "dependencies": {
   ...
  "pixi.js": "^4.3.0",
  "react-pixi": "^0.9.4",
  ...
  },
...
```

```
import React from 'react'
import {inject, observer} from 'mobx-react'
import {autorun} from 'mobx'

import pixi from 'pixi.js';
import {Stage, Text} from 'react-pixi';

import {Modal, ProgressCircular} from 'react-onsenui'

const value = ({count, size}) => Math.round(count * 100 / size);

@observer(["store"])
class LoadingModal extends React.Component{

  disposer;

  constructor() {
    super();
    this.state = {value:0}
  }

  componentDidMount(){
    this.disposer = autorun(()=>this.setState({value: value(this.props.store.ui.loadingData)}))
  }

  componentWillUnmount(){
    this.disposer();
  }

  render(){
    return (
        <Stage>
          <Text text={this.props.store.ui.loadingData.progress} x={30} y={10} key="2" />
        </Stage>
    )
  }
}

export default LoadingModal;
```

**Comment by Izzimach** [comment]
Not sure, but the problem might be with the way you import PIXI. [Check here](https://github.com/pixijs/pixi.js/issues/3204) for the explanation and see if that fixes your problem.

**Comment by Izzimach** [comment]
TLDR use

`import * as pixi from 'pixi.js';`

instead of
`import pixi from 'pixi.js';`

**Comment by dagatsoin** [comment]
Nop, I still have the same error. I also tried with React Storybook (so with Webpack) and the error is a bit different:

```
TypeError: Cannot read property 'autoDetectRenderer' of undefined
    at Object.generateDefaultRenderer (react-pixi-commonjs.js:440)
    at Object.componentDidMount (react-pixi-commonjs.js:452)
    at eval (ReactCompositeComponent.js:265)
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)
    at eval (ReactCompositeComponent.js:264)
    at CallbackQueue.notifyAll (CallbackQueue.js:76)
    at ReactReconcileTransaction.close (ReactReconcileTransaction.js:80)
    at ReactReconcileTransaction.closeAll (Transaction.js:206)
    at ReactReconcileTransaction.perform (Transaction.js:153)
    at batchedMountComponentIntoNode (ReactMount.js:126)
```

**Comment by dagatsoin** [comment]
Indeed it seems to be related to the way Pixi is imported.

Babel compiles `react-pixi-commonjs.js` as Pixi.js should have a default export:
```
this._pixirenderer = _pixi2.default.autoDetectRenderer(props.width, props.height, { view: renderelement,
```

**Comment by Izzimach** [comment]
Hm, probably fixed by 292dfa0e and 1386931 then. These haven't been published. Are you using the build from npm?  Would you be able to try to build from `master` and see if that fixes your problem?

**Comment by Izzimach** [comment]
Should probably figure out a way to test the commonjs module. Not sure I want to make a separate repo just for that though.

**Comment by dagatsoin** [comment]
It works :)
